src/flask/helpers.py

```diff
@@ -788,3 +789,23 @@ def _split_blueprint_path(name: str) -> t.List[str]:
         out.extend(_split_blueprint_path(name.rpartition(".")[0]))
 
     return out
+
+
+def abort(
+    status: t.Union[int, "Response"], *args: t.Any, **kwargs: t.Any
+) -> "te.NoReturn":
+    """Raises an :py:exc:`HTTPException` for the given status code or WSGI
+    application using the :attr:`flask.Flask.aborter` instance.
+
+    If a status code is given, it will be looked up in the list of
+    exceptions and will raise that exception.  If passed a WSGI application,
+    it will wrap it in a proxy WSGI exception and raise that::
+
+       abort(404)  # 404 Not Found
+       abort(Response('Hello World'))
+
+    .. versionchanged:: 2.2
+       Call :attr:`flask.Flask.aborter` on the :data:`~flask.current_app`
+       rather than exposing :class:`werkzeug.exceptions.abort` directly.
+    """
+    current_app.aborter(status, *args, **kwargs)  # type: ignore[misc]
```

Comment:
The typing error here is: `src/flask/helpers.py:794: error: Implicit return in function which does not return  [misc]`.

The `aborter` is an instance of the `Aborter` class, which is [properly annotated to `NoReturn`](https://github.com/pallets/werkzeug/blob/main/src/werkzeug/exceptions.py#L851-L853) on it's `__call__` function. This seems like potentially a type-checking bug, unless I'm missing something